k8s/examples/vineyard-csidriver/README.md

```diff
@@ -0,0 +1,205 @@
+
+```bash
+$ cd k8s/examples/vineyard-csidriver
+$ make docker-build && make load-images
+```
+
+To simulate the data loading/saving of the actual pipeline, we use the nfs volume to store the data. The nfs volume is mounted to the `/mnt/data` directory of the kind cluster. Then apply the data volume as follows:
+
+```bash
+$ kubectl apply -f prepare-data.yaml
+```
+
+Deploy the rbac for the kubeflow example:
+
+```bash
+$ kubectl apply -f rbac.yaml
+```
+
+Submit the kubeflow example without vineyard to the argo server:
+
+```bash
+$ for data_multiplier in 3000 4000 5000; do \
+     argo submit --watch pipeline.yaml -p data_multiplier=${data_multiplier}; \
+done
+```
+
+Clear the previous resources:
+
+```bash
+$ argo delete --all
+```
+
+Submit the kubeflow example with vineyard to the argo server:
+
+```bash
+$ for data_multiplier in 3000 4000 5000; do \
+     argo submit --watch pipeline-with-vineyard.yaml -p data_multiplier=${data_multiplier}; \
+done
+```
+
+## Clean up
+
+Delete the rbac for the kubeflow example:
+
+```bash
+$ kubectl delete -f rbac.yaml
+```
+
+Delete all argo workflow
+
+```bash
+$ argo delete --all
+```
+
+Delete the argo server:
+
+```bash
+$ kubectl delete ns argo
+```
+
+Delete the csi driver:
+
+```bash
+$ kubectl delete csidrivers.k8s.v6d.io csidriver-sample
+```
+
+Delete the vineyard cluster:
+
+```bash
+$ vineyardctl delete vineyard-cluster
+```
+
+Delete the data volume:
+
+```bash
+$ kubectl delete -f prepare-data.yaml
+```
+
+## Result Analysis
+
+The time of argo workflow execution of the pipeline is as follows:
+
+### Argo workflow duration
+
+| data_multiplier | without vineyard | with vineyard |
+| --------------- | ---------------- | ------------- |
+| 3000(8.5G)      |    186s          |      169s     |
+| 4000(12G)       |    250s          |      203s     |
+| 5000(15G)       |    332s          |      286s     |
+
+
+Actually, the cost time of argo workflow is affected by lots of factors, such as the network, the cpu and memory of the cluster, the data volume, etc. So the time of argo workflow execution of the pipeline is not stable. But we can still find that the time of argo workflow execution of the pipeline with vineyard is shorter than that without vineyard.
+
+Also, we record the whole execution time via logs. The result is as follows:
+
+### Actual execution time
+
+| data_multiplier | without vineyard | with vineyard |
+| --------------- | ---------------- | ------------- |
+| 3000            |    139.3s        |      92.3s    |
+| 4000            |    204.3s        |      131.1s   |
+| 5000            |    289.3s        |      209.7s   |
+
+According to the above results, we can find that the time of actual execution of the pipeline with vineyard is shorter than that without vineyard. To be specific, we record the write/read time of the following steps:
+
+### Write time
+
+| data_multiplier | without vineyard | with vineyard |
```

Comment:
`data_multipiler` -> `data scale`, and use data size (e.g., `xxx Mi`) in this table.

Reply:
Done.